nixos/modules/tasks/filesystems.nix

```
@@ -239,7 +239,8 @@ in
         # https://wiki.archlinux.org/index.php/fstab#Filepath_spaces
         escape = string: builtins.replaceStrings [ " " "\t" ] [ "\\040" "\\011" ] string;
         swapOptions = sw: "defaults"
-          + optionalString (sw.priority != null) ",pri=${toString sw.priority}";
+          + optionalString (sw.priority != null) ",pri=${toString sw.priority}"
+          + optionalString (sw.discardPolicy != null) ",discard${optionalString (sw.discardPolicy != "both") "=${toString sw.discardPolicy}"}";
```

```suggestion
        swapOptions = sw: concatStringsSep "," (
          [ "defaults" ]
          ++ optional (sw.priority != null) "pri=${toString sw.priority}"
          ++ optional (sw.discardPolicy != null) "discard${optionalString (sw.discardPolicy != "both") "=${toString sw.discardPolicy}"}"
        );
```
With more elements, I think it's more readable to build a list than using nested conditional strings with "," included.

---

Yes, absolutely, sorry for that awful nesting